### DISCUSSION TITLE
Remove or change GuildChannel#updateOverwrite/#createOverwrite

### DISCUSSION THREAD

**Discussion by anandre** [discordjs/discord.js#5243] [General]
**Is your feature request related to a problem? Please describe.**
`GuildChannel#createOverwrite` and `GuildChannel#updateOverwrite` are two methods that do the same thing.  Duplicate methods doing the same thing are bad.

**Describe the ideal solution**
Remove one of the methods, or change them to do what their names suggest and *only* that - create makes a new overwrite, update changes an existing one.

**Additional context**
This is a breaking change.

**Comment by anandre** [comment]
tl;dr from server chat with @almostSouji - createOverwrite will make a new entry or set it to whatever's passed, updateOverwrite will make a new entry or edit an existing one to modify only what's passed.  That's still... bad, in my opinion.  The docs are *not* clear on this and makes it confusing for users.  Could either refactor to one `setOverwrites` method as has been suggested before with an option to `replace` or `modify` in parameters, or enforce the separation of what they do per my second suggestion.

**Comment by almostSouji** [comment]
> Isn't that already what it does?

Solved this confusion on discord, as we had discussions about this misunderstanding on my part before! The run down is:

- place or replace overwrite (if already exists)
- place or update existing overwrite based on provided flags (the other flags stay the same)
